src/models/code2vec.py

```diff
@@ -0,0 +1,86 @@
+from modelforge import register_model, Model, split_strings, merge_strings
+from collections import defaultdict
+
+
+@register_model
+class Code2Vec(Model):
+    """
+    code2vec model - source code identifier embeddings.
+    """
+    NAME = "code2vec"
+
+    def construct(self, values, paths, path_contexts):
+        self._values = values
+        self._paths = paths
+        self._path_contexts = path_contexts
+
+        self._value2index = {w: i for i, w in enumerate(values)}
```

Comment:
you should not do this here, using the mapping this way means all the computation to map values/context_paths to index are done in the driver, not using Spark. You can do it in different ways, but simplest is to create the indexes after the collect, broadcast them, then use them as mappings before collecting the `(doc, [path_context_1, ...])` RDD.